components/de_amp/browser/de_amp_url_loader.cc

```diff
@@ -62,54 +64,72 @@ DeAmpURLLoader::~DeAmpURLLoader() = default;
 
 void DeAmpURLLoader::OnBodyReadable(MojoResult) {
   if (state_ == State::kSending) {
-    // The pipe becoming readable when kSending means all buffered body has
-    // already been sent.
-    ForwardBodyToClient();
+    if (bytes_remaining_in_buffer_ > 0) {
```

Comment:
Why we need to do this change?

Reply:
Previously state being Sending meant that we could end, because we only looked at one chunk. But technically you should only stop when there are no more bytes to send.